src/main/kotlin/org/jetbrains/research/testspark/tools/llm/generation/hf/HuggingFaceRequestManager.kt

```diff
@@ -0,0 +1,89 @@
+package org.jetbrains.research.testspark.tools.llm.generation.hf
+
+import com.google.gson.GsonBuilder
+import com.intellij.openapi.project.Project
+import com.intellij.util.io.HttpRequests
+import com.intellij.util.io.HttpRequests.HttpStatusException
+import org.jetbrains.research.testspark.bundles.llm.LLMMessagesBundle
+import org.jetbrains.research.testspark.core.data.ChatMessage
+import org.jetbrains.research.testspark.core.monitor.ErrorMonitor
+import org.jetbrains.research.testspark.core.progress.CustomProgressIndicator
+import org.jetbrains.research.testspark.core.test.TestsAssembler
+import org.jetbrains.research.testspark.tools.llm.SettingsArguments
+import org.jetbrains.research.testspark.tools.llm.error.LLMErrorManager
+import org.jetbrains.research.testspark.tools.llm.generation.IJRequestManager
+import org.jetbrains.research.testspark.tools.llm.generation.JUnitTestsAssembler
+import java.net.HttpURLConnection
+
+/**
+ * A class to manage requests sent to large language models hosted on HuggingFace
+ */
+class HuggingFaceRequestManager(project: Project) : IJRequestManager(project) {
+    private val url = "https://api-inference.huggingface.co/models/meta-llama/"
+    private val systemPrompt = "You are a helpful and honest code and programming assistant." +
+        " Please, respond concisely and truthfully."
```

Comment:
Please, move the string literal into `resources/properties/llm/LLMDefaults.properties`. You may access it as `LLMDefaultsBundle.get("HuggingFaceInitialSystemPrompt")`.

Reply:
Addressed in https://github.com/JetBrains-Research/TestSpark/pull/212/commits/0b00e8db3c3c474e63c765b3864c2c7991b5fa4b